File: tests/fixtures/xpath_injection/java/BaselineConstantXpath.java
// Baseline: expression is a compile-time constant.  No taint reaches
// `xpath.evaluate` so no XPATH_INJECTION finding fires.
import javax.xml.xpath.XPath;
import javax.xml.xpath.XPathConstants;
import javax.xml.xpath.XPathFactory;
import org.w3c.dom.Document;
import org.w3c.dom.NodeList;

public class BaselineConstantXpath {
    public NodeList lookup(Document doc) throws Exception {
        XPath xpath = XPathFactory.newInstance().newXPath();
        return (NodeList) xpath.evaluate("//user[@role='admin']", doc, XPathConstants.NODESET);
    }
}

File: tests/fixtures/xpath_injection/ruby/baseline_constant_xpath.rb
# Baseline: expression is a compile-time constant.  No taint reaches
# `doc.xpath` so no XPATH_INJECTION finding fires.
require 'nokogiri'

def lookup
  doc = Nokogiri::XML(File.read("users.xml"))
  doc.xpath("//user[@role='admin']")
end
